map-view/src/common/functions.ts

```diff
@@ -14,29 +22,285 @@ export function getDistanceBetweenFeatures(feature1: Feature | FeatureLike, feat
+ */
+function getBboxFilterStr(bbox?: Extent): string | undefined {
+  if (bbox) {
+    const srsName = "urn:ogc:def:crs:EPSG::3879";
+    return (
+      `<BBOX><PropertyName>location</PropertyName>` +
+      `<gml:Envelope srsName="${srsName}">` +
+      `<gml:lowerCorner>${bbox[0]} ${bbox[1]}</gml:lowerCorner>` +
+      `<gml:upperCorner>${bbox[2]} ${bbox[3]}</gml:upperCorner>` +
+      `</gml:Envelope></BBOX>`
+    );
+  }
+  return undefined;
+}
+
+/**
+ * Generates an OGC Filter Encoding XML string to exclude features that intersect
+ * with any of the provided cached polygons.
+ * This constructs a <Not><Or><Intersects>...</Intersects></Or></Not> filter,
+ * or simply <Not><Intersects>...</Intersects></Not> if only one feature.
+ *
+ * @param {Array<TurfFeature> | undefined} cachedFeatures - The array of polygons to exclude.
+ * @returns {string | undefined} The XML exclusion filter string or undefined if no features to exclude.
+ */
+function getExclusionFilterForCachedPolygons(cachedFeatures: Array<TurfFeature> | undefined): string | undefined {
+  if (!cachedFeatures || cachedFeatures.length === 0) {
+    return undefined; // No features to exclude
+  }
+
+  const srsName = "urn:ogc:def:crs:EPSG::3879"; // Needs to match WFS layer's geometry CRS
+  const spatialProperty = "location"; // The name of the geometry property in WFS layer
+
+  // Create an <Or> block for all cached polygons
+  const intersectionFilters = cachedFeatures
+    .map((feature) => {
+      // Only process Polygon or MultiPolygon geometries for intersection
+      if (feature.geometry && (feature.geometry.type === "Polygon" || feature.geometry.type === "MultiPolygon")) {
+        let gmlGeometry = "";
+        if (feature.geometry.type === "Polygon") {
+          gmlGeometry =
+            `<gml:Polygon srsName="${srsName}">` +
+            `<gml:exterior><gml:LinearRing><gml:posList>${feature.geometry.coordinates[0].map((coord) => `${coord[0]} ${coord[1]}`).join(" ")}</gml:posList></gml:LinearRing></gml:exterior>` +
+            // Add interior rings if any
+            (feature.geometry.coordinates.length > 1
+              ? feature.geometry.coordinates
+                  .slice(1)
+                  .map(
+                    (ring) =>
+                      `<gml:interior><gml:LinearRing><gml:posList>${ring.map((coord) => `${coord[0]} ${coord[1]}`).join(" ")}</gml:posList></gml:LinearRing></gml:interior>`,
+                  )
+                  .join("")
+              : "") +
+            `</gml:Polygon>`;
+        } else if (feature.geometry.type === "MultiPolygon") {
+          // MultiPolygon is more complex for direct GML representation in FE 2.0.
+          // For simplicity, converting MultiPolygon to multiple Polygon filters.
+          // A more robust solution might require creating a gml:MultiSurface.
+          gmlGeometry = feature.geometry.coordinates
+            .map(
+              (polygonCoords) =>
```

Comment:
It looks like we're doing the exact the same as the polygon case, multiple times